### PR TITLE
Default s3_deleted to false when creating new Benefits Intake Upload Submission records

### PR DESCRIPTION
**Pull request by michelpmcdonald**
## Summary

Benefits Intakes has a sidekiq job that removes consumer supplied s3 files once the file has been uploaded to EMMS for intake'in.  The job's query runs slow, we tracked it down to the query having to look for records who's s3_deleted boolean column is either nil or false.  Databases are weird, queries for nil boolean columns can be inexplicitly slow.  To avoid this, set s3_deleted to false instead of nil.  The [ticket ](https://jira.devops.va.gov/browse/API-39107) has more details.

Once this is merged, all new records will have s3_deleted defaulted to false.

Team banana-peels, we own this code

After this is merged, we'll need to:
Update any existing records with nil s3_deleted columns from nil to false, there are usually only a few 10k of them around at any given time.
Alter the database itself to default new upload submission records s3_deleted column to false.
Alter the database itself to disallow nils in the upload submission records s3_deleted column.

## Related issue(s)

[API-39107](https://jira.devops.va.gov/browse/API-39107)

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
Benefits Intake Submissions new record creations and the BI upload remover sidekiq job

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.

### CODE REVIEW

**Comment by michelpmcdonald** [comment]
moved into draft, I thought about it and figure Active Record has a way to set a default value on new records.....

File: modules/vba_documents/app/models/vba_documents/upload_submission.rb [hunk 1]
```diff
@@ -10,6 +10,9 @@ class UploadSubmission < ApplicationRecord
     include SetGuid
     include SentryLogging
     include Webhooks
+
+    attribute :s3_deleted, default: false
```
Comment (kristen-brown) [inline modules/vba_documents/app/models/vba_documents/upload_submission.rb:14]: Note: Unlike a data migration, this won't default `s3_deleted` to `false` for previously created records. Those will still be `s3_deleted: nil`.

File: modules/vba_documents/app/sidekiq/vba_documents/upload_remover.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ class UploadRemover
 
     REMOVAL_QUERY = <<-SQL.squish
         status IN ('received', 'processing', 'error', 'success', 'vbms')
-        AND s3_deleted IS NOT True
+        AND s3_deleted IS False
```
Comment (kristen-brown) [inline modules/vba_documents/app/sidekiq/vba_documents/upload_remover.rb:17]: I tested this query out against my local database, which had four records in "vbms" status, with the following `s3_deleted` values: `true`, `nil`, `false`, `false`. The record with `nil` was created prior to pulling this branch locally. When I tested the query, it returned only two records, the `false` ones. (Note: I temporarily removed the date check from the query to simply my test.) 

Therefore, I think there's an unintended consequence of this change, that for records created before this PR is merged (where the `attribute` default is added), they will not be caught by this job, and therefore not removed from S3.

**Review by kristen-brown — CHANGES_REQUESTED** [review]

File: modules/vba_documents/app/models/vba_documents/upload_submission.rb (hunk 1 above)
Comment (michelpmcdonald) [reply modules/vba_documents/app/models/vba_documents/upload_submission.rb:14]: yep, two options here, have the upcoming migration change any nils to false, or, as I was leaning towards, just manually change them myself after this PR is deployed.

Having the migration do the data changes is already hitting some issue, they have some kind of migration performance bot in place that encourages a manual change.

Comment (kristen-brown) [reply modules/vba_documents/app/models/vba_documents/upload_submission.rb:14]: Sorry, I probably missed your comment!

File: modules/vba_documents/app/sidekiq/vba_documents/upload_remover.rb (hunk 2 above)
Comment (kristen-brown) [reply modules/vba_documents/app/sidekiq/vba_documents/upload_remover.rb:17]: Yes, sorry, I missed your comment about having a separate PR or process for the older records.

**Review by kristen-brown — APPROVED** [review]

**Review by mathisto — APPROVED** [review]
Looks great; noting the additional PR for the old records.
seal-of-approval-thumbs-up-gif

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**